src/internal/connector/onedrive/collections.go

```diff
@@ -427,15 +437,46 @@ func (c *Collections) Get(
 
 	observe.Message(ctx, fmt.Sprintf("Discovered %d items to backup", c.NumItems))
 
-	// Add an extra for the metadata collection.
 	collections := []data.BackupCollection{}
 
+	// add all the drives we found
 	for _, driveColls := range c.CollectionMap {
 		for _, coll := range driveColls {
 			collections = append(collections, coll)
 		}
 	}
 
+	drivePathPfx, err := c.servicePathPfxFunc()
+	if err != nil {
+		return nil, clues.Wrap(err, "making service prefix for drive tombstone").WithClues(ctx)
+	}
+
+	// generate tombstones for drives that were removed.
+	for driveID := range driveTombstones {
+		prevDrivePath, err := drivePathPfx.Append("drives", driveID, "root:").AsDataLayerPath(false)
```

Comment:
I'm a bit conflicted about adding another function to the `path` package since it already has so many. I know path manipulation has become one of the things we do often in Corso, but we should still be careful about how we allow manipulating and building them. While this works, it feels a little roundabout to go path.Path -> path.Builder -> path.Path when we should be able to just go from a prefix path to a longer path

Probably in the long run we should expand how `path.Path.Append` works to allow multiple elements to be appended. Doing so will generate some churn though as there seem to be ~25 uses of that call in the codebase right now

as an interim solution, would it be alright to change `servicePathPfxFunc` to instead take args and build the whole path when called? Perhaps it could be something like
```go
// earlier, setup c.servicePathPfxFunc
c.servicePathPfxFunc := func(driveID string) (path.Path, error) {
  // Prefix path args bound during function creation time.
  return path.Build(
    tenant,
    resourceOwner,
    service,
    cat,
    false,
    // Sufficient to just delete the drive ID folder since delete
    // is recursive.
    "drives",
    driveID)
}

for driveID := range driveTombstones {
  prevDrivePath, err := c.servicePathPfxFunc(driveID)

  ...
}
```

Reply:
Sure, I think that's a reasonable short-term fix.